### DISCUSSION TITLE
GPL Licensing

### DISCUSSION THREAD

**Discussion by jakehader** [PySlurm/pyslurm#292] [Q&A] [answered]
Hi there,

I apologize if this is not a good place to put this discussion, but I am wondering if licensing under GPL is required for this project? Our team would like to use this project for some applications based on our nuclear reactor design framework (https://github.com/terrapower/armi), but much of our code cannot be open sourced due to a variety of reasons. I see from https://github.com/orgs/PySlurm/discussions/279 that this project is getting some new renovations that we are excited about, but we are trying to weigh our options for using `PySLURM` directly versus building some new tooling based on the SLURM Rest API (https://slurm.schedmd.com/rest.html). 

I am not certain on how/if licenses can just be "changed" retroactively, but I figured I'd ask in case. I'd love to be able to leverage the great work here and not reinvent any wheels. For our specific project we are typically looking at MIT or Apache 2.0 Licenses for third-party software.

Thank you!

**Comment by giovtorres** [comment, marked as answer]
It is my understanding that any program that links to `libslurm` automatically becomes GPL. Therefore, PySlurm would need to be GPL as well.

**Reply by jakehader** [reply]
I see that the underlying SLURM API libraries are GPL now so this makes sense. Thanks for the quick response!
